libraries/provider_pg_repluser.rb

```diff
@@ -0,0 +1,25 @@
+class Chef
+  class Provider
+    # adds replication user to database
+    class PgRepluser < Chef::Provider::LWRPBase
+      use_inline_resources if defined?(use_inline_resources)
+
+      def whyrun_supported?
+        true
+      end
+
+      action :create do
+        execute 'set-replication-user' do # ~FC009
+          role_exists = %(psql -c "SELECT rolname FROM pg_roles WHERE rolname='#{new_resource.repl_user}'" | grep #{new_resource.repl_user})
```

Comment:
Have you considered doing this in ruby rather than shelling out, perhaps with the `pg` gem, along the lines of:
```
require 'pg'
psql = PG.connect( dbname: 'postgres' )
psql.exec("select rolname from pg_roles where rolname='#{new_resource.repl_user}'")
```